src/emulator/cpu.ts

```diff
@@ -109,26 +109,76 @@ export class Cpu {
         this.interrupt.clear(irq);
         this.state.interruptsEnabled = false;
 
-        this.state.r16[r16.sp] = (this.state.r16[r16.sp] - 1) & 0xffff;
-        this.bus.write(this.state.r16[r16.sp], this.state.p >>> 8);
-        this.state.r16[r16.sp] = (this.state.r16[r16.sp] - 1) & 0xffff;
-        this.bus.write(this.state.r16[r16.sp], this.state.p & 0xff);
+        this.stackPush(this.state.p);
 
         this.state.p = getIrqVector(irq);
         this.clock.increment(5);
 
         return 5;
     }
 
+    private stackPush(value: number): void {
```

Comment:
Maybe rename that to `stackPush16` for clarity? Although there is no 8 bit version on the gameboy, so 🤷‍♂️